### ISSUE TITLE
Feature request: Activate inactive tool on change

### ISSUE THREAD

**Issue by rebio** [Beep6581/RawTherapee#6363]
The idea is very simple and straightforward: If you change any slider, any drop down menu or any number in the tool, the tool should be activated,

The usage would be to make the editing process smoother. This would be especially good to have for the Local adjustments.
Many new users will forget to turn the tab on or maybe have no idea why it would not work.

Since I can image that some people/pro users would not like that, you should be able to toggle it in the settings.

**Comment by ghost** [comment]
The tools not automatically being activated when making changes is part of my workflow.  This is true for certain modules, but especially so for the local adjustments section.

I would not like to see this implemented without the above mentioned option in the preferences to disable it. I would even go as far as to say that this option should not be enabled by default.

I don't think that *(new) people don't know how to do X* is a valid argument: People should read/check/enquire RawPedia. 

Just my 2c on this subject.

**Comment by rebio** [comment]
> I would not like to see this implemented without the above mentioned option in the preferences to disable it. I would even go as far as to say that this option should not be enabled by default.
> 
> I don't think that _(new) people don't know how to do X_ is a valid argument: People should read/check/enquire RawPedia.

Yes, RawPedia is a good guide for RT - but to be fair: It's simply too much for new users who don't know what they are actually looking for.
Just telling them _Go look it up in RawPedia_ should also not be the single way to go. Kinda like telling a new users to go for Arch when they are interested in Linux. For sure that will teach them a lesson, but the lesson would more than likely be to stay away from the tool.

You certainly should not make everything as simple as possible to please the most inexperienced user. However, advanced users emerge out of beginners. So not being _entry level hostile_ is not a bad thing as such. Of course advanced users have a different workflow and what I usually like about FOSS is that you can configure it to your needs.

Just a thought on this: LibreOffice has a little window on the first startup that let's you select the look you want to have. Many Linux distros have options where you can select keyboard, mice, time & region settings on startup and install. I know I'm kinda driftig away from the actual point, but what about some sort of welcome screen on the first start that let's you select the setup and maybe some important configs (external editor, save path etc.) with presets. That would solve such _I want it that way_ problem: Just select it on install.

**Comment by ghost** [comment]
I'm not trying to be the bad guy here :sweat_smile: 

I'm not *entry level hostile*, I have, however, come across too many new users that did not even glance at RawPedia when they run into something that doesn't work the way they expect it to. So, I do expect new users to, at least, read the _General Information/Comments_ section(s). And, yes, I do have a problem with turning RawTherapee into a *for dummies* state out-of-the-box.

I am more then willing to help newbies and never just tell them to _Go look it up in RawPedia_; I answer the question **and** point to the more elaborate RawPedia article if one is present **and** point them to outside information (think: Andy Astbury's videos for example). So my previous comment isn't meant to be newbie hostile in any way.

Being blunt: RawTherapee isn't an entry level RAW editor and I don't see the point in dumbing it down to accommodate a small section of new users.

Again, just my 2c.

**Comment by arturoisilvia** [comment]
Hi,
I also don't want the module to activate when you move a slider, because I have a slow pc and with every change in a slider I have to wait, then I go to the next slider, change it and have to wait, then the next one... and so on until the end and desperately it's a long time. 
As it is now, I change all the sliders in the module, activate it and only have to wait once.

**Comment by rebio** [comment]
> I'm not trying to be the bad guy here 😅

I don't want my comment to be understood as finger pointing or blaming anyone doing anything that encourages a hostile environment. I don't think that such accusations help anyone anywhere with anything.

The truth is that people are more easily discouraged than encouraged. Something that is a trivial thing for an advanced user to do may is a challenge for a new user. I can understand that the advanced user may would be annoyed by a new setting or a usecase that is so clearly written in RawPedia.

If RT was a high mountain, a professional mountain climber would have no problem getting to the top of it. Though, why wouldn't you provide an easy route for beginners? The route is not simply _RawPedia_. RawPedia is more like the trail that leads upward, but there is much more to climbing than just the trail itself.

> I have, however, come across too many new users that did not even glance at RawPedia when they run into something that doesn't work the way they expect it to.

What if there was a 'Simple Mode' in RT that links different tools in RT to sections in RawPedia.
Just to be clear on this: Even mods in Minecraft usually have a better tool-to-description link than RT. That is something that could be expanded.

> And, yes, I do have a problem with turning RawTherapee into a _for dummies_ state out-of-the-box.

RT is delivered with 9 different UI skins. 8 of them are not in use. That's also not a problem, is it. It's not a problem because they are hidden, right? Who says a setting would have to be an _in your face_ style either? I'm not asking for a _for dummies_ layout right from the start.
I'm asking for a setting that I would find very useful. I'm asking for a toggle because I can imagine some people (eg.  
arturoisilvia) would not like that setting to be active for one or another reason. I'm asking to not hide such a toggle deep inside a menu where it is never to be seen again.

> Being blunt: RawTherapee isn't an entry level RAW editor and I don't see the point in dumbing it down to accommodate a small section of new users.

From RTs main page: "The target audience ranges from enthusiast newcomers [...] to semi-professional photographers".
I do NOT suggest that RTs goal should be to please the low end tick-tock und instagram user who is not interested in image editing that is beyond selecting a filter. But I do suggest to not leave out the enthusiast newcomer.

Sidenote: Would such a setting and it's toggle really be that hard to implement that justifies such a long discussion?

**Comment by ghost** [comment]
> Would such a setting and it's toggle really be that hard to implement [...]

I don't know how hard it would be to implement this *Activate inactive tool on change* request. My developer skills are limited to (bash/ksh) shell scripting and some perl and I'm not familiar with RT's C++ based code. 
 
I do know it's more then just an on/off setting in the preferences menu, though. This request would touch just about every single slider, curve, button and free field in the RT editor section. There should be an "intelligence", for a better word, to this setting too. What to do if: I move a slider from its default position, which would automatically turn on the tool, and then turn off the tool manually. This should be allowed and a circular  condition should be avoided. Does RT need to turn the tool back on automatically when that, or another slider, is moved (maybe even back to its default position). By now you have a lot of extra code.

This might also be considered as being not all that important compared to the other bugs/requests that are still open, especially because turning a module or tool on/off by hand is easily done. Unfortunately RT does have a rather small developer base so not everything gets to be picked up and, if it turns out to be a good idea, implemented.

**Comment by rebio** [comment]
> Hi, I also don't want the module to activate when you move a slider, because I have a slow pc and with every change in a slider I have to wait, then I go to the next slider, change it and have to wait, then the next one... and so on until the end and desperately it's a long time. As it is now, I change all the sliders in the module, activate it and only have to wait once.

I had an idea for this: What about a toggle setting that let's you select between constant recalculation or calculation on button press. Kinda like there already is in the queue tab where you can set the queue to start automatically.

The button for this could be located next to the progress bar at the bottom of the editor. So you can change every setting to your liking without any further calculation that would slow down your PC.
